apps/antalmanac/src/lib/enrollmentHistory.ts

```diff
@@ -0,0 +1,154 @@
+// and waitlistHistory arrays into one array that contains the enrollment data
+// for each day
+export interface EnrollmentHistory {
+    year: string;
+    quarter: string;
+    department: string;
+    courseNumber: string;
+    days: EnrollmentHistoryDay[];
+    instructors: string[];
+}
+
+export interface EnrollmentHistoryDay {
+    date: string;
+    totalEnrolled: number;
+    maxCapacity: number;
+    waitlist: number | null;
+}
+
+class _EnrollmentHistory {
+    // Each key in the cache will be the department and courseNumber concatenated
+    enrollmentHistoryCache: Record<string, EnrollmentHistory>;
+    termShortNames: string[];
+
+    constructor() {
+        this.enrollmentHistoryCache = {};
+        this.termShortNames = termData.map((term) => term.shortName);
+    }
+
+    queryEnrollmentHistory = async (department: string, courseNumber: string): Promise<EnrollmentHistory | null> => {
+        const cacheKey = department + courseNumber;
+        if (cacheKey in this.enrollmentHistoryCache) {
+            return this.enrollmentHistoryCache[cacheKey];
+        }
+
+        // Query for the enrollment history of all lecture sections that were offered
+        const queryString = `{
+            enrollmentHistory(department: "${department}", courseNumber: "${courseNumber}", sectionType: Lec) {
+                year
+                quarter
+                department
+                courseNumber
+                dates
+                totalEnrolledHistory
+                maxCapacityHistory
+                waitlistHistory
+                instructors
+            }
+        }`;
+
+        const res =
+            (await queryGraphQL<EnrollmentHistoryGraphQLResponse>(queryString))?.data?.enrollmentHistory ?? null;
+
+        if (res && res.length > 0) {
+            const parsedEnrollmentHistory = this.parseEnrollmentHistoryResponse(res);
+            this.sortEnrollmentHistory(parsedEnrollmentHistory);
+
+            // For now, just return the enrollment history of the most recent quarter
+            // instead of the entire array of enrollment histories
+            const latestEnrollmentHistory = parsedEnrollmentHistory[0];
+            this.enrollmentHistoryCache[cacheKey] = latestEnrollmentHistory;
+            return latestEnrollmentHistory;
+        }
+
+        return null;
+    };
+
+    /**
+     * This function parses enrollment history data from PeterPortal so that
+     * we can pass the data into a recharts graph. For each element in the given
+     * array, merge the dates, totalEnrolledHistory, maxCapacityHistory,
+     * and waitlistHistory arrays into one array that contains the enrollment data
+     * for each day.
+     *
+     * @param res an array of enrollment histories from PeterPortal
+     * @returns an array of enrollment histories that we can use for the graph
+     */
+    parseEnrollmentHistoryResponse = (res: EnrollmentHistoryGraphQL[]): EnrollmentHistory[] => {
```

Comment:
This should be a static method or global function.
```suggestion
    static parseEnrollmentHistoryResponse (res: EnrollmentHistoryGraphQL[]): EnrollmentHistory[] {
```